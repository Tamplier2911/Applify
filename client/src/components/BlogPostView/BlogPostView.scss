.blog-post-view {
  cursor: pointer;
  display: grid;
  grid-template-rows: repeat(2, 1fr);

  // background-position: center;
  // background-size: 180%;
  // background-repeat: no-repeat;
  // backface-visibility: hidden;

  color: var(--cl-white);
  text-decoration: none;
  font-weight: 300;
  font-family: var(--font-title);

  // min-height: 40rem; // double check
  box-shadow: 0 0.2rem 0.5rem var(--cl-square-shadow);

  // transition: background-size 8s, filter 0.5s, box-shadow 0.3s;
  transition: filter 0.5s, box-shadow 0.3s;
  animation: 1s blogFadeIn;
  filter: brightness(90%);

  &:hover {
    filter: brightness(130%);
    // background-size: 140%;
    box-shadow: 0 0.2rem 1rem var(--cl-card-shadow-hover);
  }

  &__image {
    grid-row: 1 / -1;
    grid-column: 1 / -1;
    width: 100%;
    height: 100%;
    display: block;
    object-fit: cover;
  }

  &__content {
    grid-row: 2 / 3;
    grid-column: 1 / -1;
    display: grid;
    background-color: rgba(0, 0, 0, 0.432);
    grid-template-columns: repeat(2, 1fr);
    grid-template-columns: 1fr min-content;

    &--wrapper {
      display: grid;
      grid-auto-rows: min-content;
      align-content: center;
      padding: 1rem 2rem;
    }
    &--theme {
      font-size: 1.5rem;
    }
    &--title {
      font-weight: 400;
      font-size: 1.7rem;
    }
    &--date {
      font-size: 1.5rem;

      display: grid;
      grid-template-columns: 1fr min-content;
      align-items: center;

      &-itself {
        align-items: center;
      }
    }

    &--likewrap {
      display: grid;
      grid-template-columns: repeat(2, 1fr);
      grid-column-gap: 0.5rem;
      align-items: end;
    }
    &--svg {
      width: 2rem;
      height: 2rem;
      fill: var(--cl-white);
      transition: fill 0.3s;
      &:hover {
        fill: #eb4646;
      }
    }
    &--like {
      font-size: 1.5rem;
    }
  }
}

.big {
  grid-column: span 2;
  @media only screen and (max-width: 425px) {
    grid-column: 1 / 2;
  }
}

.small {
  grid-column: span 1;

  // @media only screen and (max-width: 990px) {
  //   grid-column: span 2;
  // }

  @media only screen and (max-width: 768px) {
    grid-column: span 2;
  }

  @media only screen and (max-width: 425px) {
    grid-column: 1 / 2;
  }
  //   background-color: floralwhite;
}

.liked {
  fill: #eb4646;
}

@keyframes blogFadeIn {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
