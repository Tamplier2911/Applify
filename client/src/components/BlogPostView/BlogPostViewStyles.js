import styled, { css } from "styled-components";
import { Link } from "react-router-dom";

// svg
import { ReactComponent as HartSVG } from "../../assets/svg/heart.svg";

const contentOverflowFix = css`
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
`;

const blogFadeInAnimation = css`
  @keyframes blogFadeIn {
    0% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }
`;

const getLikeColor = (props) => {
  const { liked } = props;
  if (liked) {
    return `fill: var(--cl-bpost-liked);`;
  } else {
    return `fill: var(--cl-white);`;
  }
};

const bigOrSmall = (props) => {
  const { big } = props;
  if (big) {
    return `grid-column: span 2;
            @media only screen and (max-width: 425px) {
                grid-column: 1 / 2;
            }`;
  } else {
    return `grid-column: span 1;
            @media only screen and (max-width: 768px) {
                grid-column: span 2;
            }
            @media only screen and (max-width: 425px) {
                grid-column: 1 / 2;
            }`;
  }
};

export const BlogViewContainer = styled(Link)`
  ${bigOrSmall}
  ${blogFadeInAnimation}
  cursor: pointer;
  display: grid;
  grid-template-rows: repeat(2, 1fr);

  color: var(--cl-blogview-desc);
  text-decoration: none;
  font-weight: 300;
  font-family: var(--font-title);

  min-height: 20rem; // double check
  max-height: 20rem; // double check
  box-shadow: 0rem 0rem 0.5rem var(--cl-square-shadow);

  transition: filter 0.5s, box-shadow 0.3s;
  animation: 1s blogFadeIn;
  filter: brightness(90%);

  @media only screen and (max-width: 425px) {
    grid-template-rows: repeat(10, 1fr);
    max-height: 30rem;
  }

  &:hover {
    filter: brightness(130%);
    // box-shadow: 0rem 0rem 1rem var(--cl-card-shadow-hover);
    box-shadow: 0rem 0rem 1rem var(--cl-primary);
  }
`;

export const BlogViewImg = styled.img`
  grid-row: 1 / -1;
  grid-column: 1 / -1;
  width: 100%;
  height: 100%;
  display: block;
  object-fit: cover;
`;

export const BlogViewContent = styled.div`
  grid-row: 2 / 3;
  grid-column: 1 / -1;
  display: grid;
  background-color: var(--cl-bpost-view-shadow);
  grid-template-columns: repeat(2, 1fr);
  grid-template-columns: 1fr min-content;

  @media only screen and (max-width: 425px) {
    grid-row: 8 / 11;
  }
`;

export const BlogViewContentWrap = styled.div`
  display: grid;
  grid-auto-rows: min-content;
  align-content: center;
  padding: 1rem 2rem;
`;

export const BlogViewContentTheme = styled.div`
  font-size: 1.5rem;
  ${contentOverflowFix}
`;

export const BlogViewContentTitle = styled.h4`
  font-weight: 400;
  font-size: 1.7rem;
  ${contentOverflowFix}
`;

export const BlogViewContentDateWrap = styled.div`
  font-size: 1.5rem;
  display: grid;
  grid-template-columns: 1fr min-content;
  align-items: center;
`;

export const BlogViewContentDate = styled.div`
  align-items: center;
`;

export const BlogViewContentLikeWrap = styled.div`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-column-gap: 0.5rem;
  align-items: center;
`;

export const BlogViewContentLike = styled.div`
  font-size: 1.5rem;
`;

export const BlogViewContentLikeSVG = styled(HartSVG)`
  ${getLikeColor}
  width: 2rem;
  height: 2rem;
  transition: fill 0.3s;
  &:hover {
    fill: var(--cl-bpost-like-hover);
  }
`;
