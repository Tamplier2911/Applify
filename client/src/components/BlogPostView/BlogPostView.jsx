import "./BlogPostView.scss";
import React from "react";
import { Link } from "react-router-dom";

// JS Rendering CSS
import {} from "./BlogPostViewStyles";

// svg
import { ReactComponent as HartSVG } from "../../assets/svg/heart.svg";

const BlogPostView = ({ id, title, theme, date, like, image, liked, pos }) => {
  const big = [0, 3, 4];
  //   const small = [1001, 1002, 1005];

  /*
  <Link
      to={`${id}`}
      className={`blog-post-view ${big.includes(pos) ? "big" : "small"}`}
      style={{
        backgroundImage: `linear-gradient(to bottom, transparent , #0000007e 40%) ,url(${image})`
      }}>
  </Link>
  */

  return (
    <Link
      to={`blog/${id}`}
      className={`blog-post-view ${big.includes(pos) ? "big" : "small"}`}
    >
      <img
        className="blog-post-view__image"
        alt="blog post presentation"
        src={`${image}`}
      ></img>
      <div className="blog-post-view__content">
        <div className="blog-post-view__content--wrapper">
          <div className="blog-post-view__content--theme">{theme}</div>
          <h4 className="blog-post-view__content--title">{title}</h4>
          <div className="blog-post-view__content--date">
            <div className="blog-post-view__content--date-itself">
              {date.toLocaleString("en-us", {
                day: "numeric",
                month: "long",
                year: "numeric"
              })}
            </div>
            <div className="blog-post-view__content--likewrap">
              <div className="blog-post-view__content--like">{like}</div>
              <HartSVG
                className={`blog-post-view__content--svg ${
                  liked ? "liked" : "none"
                }`}
              />
            </div>
          </div>
        </div>
      </div>
    </Link>
  );
};

/*
console.log(
    blogs[0].date.toLocaleString("en-us", {
      day: "numeric",
      month: "long",
      year: "numeric"
    })
  );
*/

export default BlogPostView;
