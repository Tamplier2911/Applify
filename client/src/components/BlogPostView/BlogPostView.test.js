import React from "react";
import toJson from "enzyme-to-json";

// component
import BlogPostView from "./BlogPostView";

// utils
import {
  setShallowC,
  findByTestAttr,
  storeFactory,
  checkProps,
} from "../../tests/testUtils";

describe("<BlogPostView />", () => {
  const initialStore = {
    user: {
      userObject: {
        id: "",
        name: "",
        email: "",
        photo: "",
        about: "",
        likedBlogposts: [],
      },
      errorMessage: null,
      isLogged: false,
    },
  };
  const store = storeFactory(initialStore);

  let wrapper;
  beforeEach(() => {
    wrapper = setShallowC(<BlogPostView store={store} />);
  });

  it("renders without an error", () => {
    const blogPostView = findByTestAttr(wrapper, "blog-post-view");
    expect(blogPostView.length).toBe(1);
  });

  it("renders children with content content", () => {
    const blogPostViewContent = findByTestAttr(
      wrapper,
      "blog-post-view-content"
    );
    expect(blogPostViewContent.length).toBe(1);
  });

  it("recieves expected props", () => {
    const expectedProps = {
      blog: {
        author: { photo: "str", _id: "str", name: "str" },
        content: "str",
        createdAt: "str",
        image: "str",
        likes: 0,
        theme: "str",
        title: "str",
      },
      pos: 0,
      slot: 0,
      index: 0,
    };
    const err = checkProps(BlogPostView, expectedProps, "prop");
    expect(err).toBeUndefined();
  });

  it("matches snapshot", () => {
    const blogPostView = findByTestAttr(wrapper, "blog-post-view");
    expect(toJson(blogPostView)).toMatchSnapshot();
  });
});
