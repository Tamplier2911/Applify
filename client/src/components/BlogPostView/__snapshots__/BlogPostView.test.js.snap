// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogPostView /> matches snapshot 1`] = `
<Styled(Link)
  big={0}
  data-test="blog-post-view"
  to="blog/undefined-undefined-undefined"
>
  <styled.img
    alt="blog post presentation"
    src="http://localhost/api/uploads/images/users/default.jpg"
  />
  <styled.div
    data-test="blog-post-view-content"
  >
    <styled.div>
      <styled.div />
      <styled.h4 />
      <styled.div>
        <styled.div>
          August 29, 2020
        </styled.div>
        <styled.div
          onClick={[Function]}
        >
          <styled.div />
          <Styled(Component)
            liked={0}
          />
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
</Styled(Link)>
`;
